addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -260,6 +260,14 @@ export class PaymentScreen extends Component {
         if (!this.check_cash_rounding_has_been_well_applied()) {
             return;
         }
+        const linesToRemove = this.currentOrder.lines.filter((line) => {
+            const rounding = line.product_id.uom_id.rounding;
+            const decimals = Math.ceil(-Math.log10(rounding));
```

Comment:
```suggestion
            const decimals = Math.max(0, Math.ceil(-Math.log10(rounding)));
```

Let's make an assumption here that `decimals` should always be greater or equal than zero.